dtcli/delete_extension.py

```diff
@@ -0,0 +1,116 @@
+import json
+from collections import defaultdict
+from typing import Dict, Set, Optional, List
+from pathlib import Path
+
+
+from dtcli.api import DynatraceAPIClient
+
+class State:
+    def __init__(self, d):
+        self.d = d
+
+    def __getitem__(self,key):
+        return self.d[key]
+
+    def __contains__(self, key):
+        return key in self.d
+
+    def __str__(self):
+        return str(self.d)
+
+    def versions(self, extension_fqdn, exclude: Optional[Set[str]] = None) -> List[str]:
+        if exclude is None:
+            exclude = set()
+
+        all_versions: Set[str] = set(self[extension_fqdn].keys())
+
+        return list(sorted(all_versions - exclude))
+
+    def as_dict(self):
+        return self.d
+
+def acquire_state(client: DynatraceAPIClient) -> State:
```

Comment:
I think I've mentioned the idea of only downloading the data that we actually need for wiping, is there an issue associated with it or a TODO?